igor-monitor-travis/src/main/java/com/netflix/spinnaker/igor/travis/service/TravisService.java

```diff
@@ -338,6 +346,19 @@ public Build getBuild(String repoSlug, int buildNumber) {
         .collect(Collectors.toList());
   }
 
+  private List<V3Build> getBuildsForSpecificRepos(Collection<String> repositories) {
+    return repositories
+        .parallelStream()
+        .map(
+            repoSlug -> {
+              log.info("fetching builds for repo: {}", repoSlug);
+              return travisClient.v3builds(
+                  getAccessToken(), repoSlug, buildResultLimit, addLogCompleteIfApplicable());
+            })
+        .flatMap(builds -> builds.getBuilds().stream())
+        .collect(Collectors.toList());
```

Comment:
How do exceptions get handled in a `parallelStream`? Will it abort the whole stream (I would assume so). Would you prefer to abort from the entire op, or get a partial response?

Reply:
Good point, @robzienert. Made the changes to skip the repo if it fails.